records/2022-07-20.security.prefix-api-token.md

```diff
@@ -0,0 +1,39 @@
+# TITLE
+
+Date: 2022-07-20
+
+## Status
+
+**DRAFT**
+
+## Context
+
+The mishandling of application secrets has lead to many security incidents in companies big and small. Given the inherent risk in how users store these secrets, it is important that we provide mechanisms to protect our users from doing so improperly.  Taking a cue from [Github](https://github.blog/2021-04-05-behind-githubs-new-authentication-token-formats/), [Slack](https://api.slack.com/authentication/token-types), and [Stripe](https://stripe.com/docs/api/authentication), we see that token prefixes are an effective technique to help us ensure our tokens don't end up in public repositories.
+
+By combining a known prefix with Github's [secret scanning program](https://docs.github.com/en/developers/overview/secret-scanning-partner-program), we can be alerted if one of our users mistakenly stores their API keys in the open and we can take action.
```

Comment:
Thanks for linking in github documentation and being exhaustive. Do we know if there are other public repositories that could host our public keys as well? Should we write down as well that private governmental repositories are considered safe? We should ask these questions to the SRE team.

Reply:
I've added a [list of other public repos](https://github.com/cds-snc/notification-adr/pull/24/files#diff-2390265d8bdcdf72117c643d55fd9049c8b400c79f0a1a76d97d293e3513f79aR50-R55).  @mohamed-cds any others I should add?

Reply:
i wouldn't suggest that private gov repos are safe since at any point in the future they may make their repos public. 

The recommendation is to treat private repos the same as public incase there's a vulnerability in the future that grants outsiders access to the source code. e.g. Heroku breach that lead to Github tokens, which granted access to private repos.

@andrewleith github secret scanning is our first attempt at this, so what you have no + future considerations works well.